### PR TITLE
[Keyboard] Convert Corne Keyboard to Split Common

### PR DESCRIPTION
**Pull request by drashna**
## Description

Converts the Corne Keyboard to use the Split Common keyboard code. 

@foostan are you okay with this, and do you have a preference? 

This breaks reactive modes for RGB Matrix on the board until Split Common is changed to support it (eg #5998 ), as well as OLED support on the secondary/slave half.  

I don't feel comfortable having this merged until 5998 or something similar makes it way into split common.

## Types of Changes
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
This PR should be merged AFTER #5998

**Comment by dsanchezseco** [comment]
I was trying to do the same with my corne, guess now I know why the OLED wasn't updating in the slave when changing layer (not using I2C yet).

**Comment by drashna** [comment]
@daisukekoba sorry for not replying sooner! 

The reason for the OLED not working properly may be because it doesn't detect the "change" properly.  I think. And it may need custom handling to work right. 

As for the rgb matrix. no, that's mostly unchanged.

**Comment by rpbaptist** [comment]
I have been testing this with #5998 and I noticed the RGB matrix is being mirrored from master to slave side.

**Comment by drashna** [comment]
@rpbaptist  Remove `isLeftHand` check from the `rev1.c` file, that should fix the issue.

**Comment by rpbaptist** [comment]
All that appears to do is disable RGB for slave side entirely while still having mirror effect with heatmap for example.

File: keyboards/crkbd/keymaps/OLED_sample/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/OLED_sample/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

File: keyboards/crkbd/keymaps/OLED_sample/rules.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/OLED_sample/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
```

File: keyboards/crkbd/keymaps/OLED_sample/rules.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = no        # Audio control and System control(+450)
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/OLED_sample/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = no        # Audio control and System control
```

File: keyboards/crkbd/keymaps/OLED_sample/rules.mk [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+CONSOLE_ENABLE = no         # Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/OLED_sample/rules.mk]:
```suggestion
CONSOLE_ENABLE = no         # Console for debug
```

File: keyboards/crkbd/keymaps/OLED_sample/keymap.c [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+  ),
+
+  [_LOWER] = LAYOUT(
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,     KC_5,                      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
+    CTLTB,   KC_F1,   KC_F2,   KC_F3,   KC_F4,    KC_F5,                     KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  XXXXXXX,
+    KC_LSFT, KC_F11,  KC_F12,  KC_F13,  KC_F14,   KC_F15,                    KC_F16,  KC_F17,  KC_F18,  KC_F19,  KC_F20,  XXXXXXX,
+                                        GUIEI,    LOWER,   KC_SPC,  KC_ENT,  RAISE,   ALTKN
+  ),
+
+  [_RAISE] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/OLED_sample/keymap.c]:
```suggestion
  [_RAISE] = LAYOUT(
```

File: keyboards/crkbd/keymaps/OLED_sample/keymap.c [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+    KC_ESC,  KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,                   KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
+    CTLTB,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   KC_MINS, KC_EQL,  KC_LCBR, KC_RCBR, KC_PIPE, KC_GRV,
+    KC_LSFT, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   KC_UNDS, KC_PLUS, KC_LBRC, KC_RBRC, KC_BSLS, KC_TILD,
+                                        GUIEI,   LOWER,   KC_SPC,  KC_ENT,  RAISE,   ALTKN
+
+  ),
+
+  [_ADJUST] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/OLED_sample/keymap.c]:
```suggestion
  [_ADJUST] = LAYOUT(
```

**Comment by rpbaptist** [comment]
After the recent updates and merging new branches everything seems to be in order. I didn't have to adjust anything in `rev1.c`

**Comment by Curry** [comment]
@drashna with the addition of #7915, You need to add 
`#define DIODE_DIRECTION COL2ROW` to `rev1/config.h`, otherwise all keymap builds will fail.

**Comment by foostan** [comment]
@drashna Thank you for your support and I'm sorry that I not contribute about it.
Would you tell me the current status of this PR?
We should wait https://github.com/qmk/qmk_firmware/pull/5998?

I'm preparing to support for the VIA configurator so I converted to split common and oled driver for each keymaps of crkbd(https://github.com/qmk/qmk_firmware/compare/master...foostan:crkbd_split_common_via).
If this PR is active and waits someone then I wait for merged this PR.

**Comment by drashna** [comment]
@foostan no worries! the board is amazing, so thank you for that!  It's been my daily driver for quite a while, as well as heavily used for development.  

It's mostly waiting on a way to handle mirroring the keymap matrix. so 5998 or the like. Otherwise, it breaks a lot of the OLED code, and it some of the rgb stuff. Which is why I've hesitated. 

But if you're okay with converting the board to split common, I think I'll update this to support both split common and the older (current) method.  This way, we can have both versions for the firmware. 

And at the same time, I can incorporate the via changes, if you're okay with that.

**Comment by foostan** [comment]
@drashna Thank you so much! I'm happy to have use it for development.

> But if you're okay with converting the board to split common, I think I'll update this to support both split common and the older (current) method. This way, we can have both versions for the firmware.
>
> And at the same time, I can incorporate the via changes, if you're okay with that.

I agree it. Would you incorporate the via changes please?

**Comment by drashna** [comment]
> I agree it. Would you incorporate the via changes please?

I'm ... mostly done with this, and will incorporate the VIA changes, once I got it compiling.

**Comment by foostan** [comment]
Thanks!

**Comment by drashna** [comment]
Okay, I think I got all of it changed and updated.  Right now, as is, it should be transparent to anyone using the board.  As well as the split common changes. 

And it should support both the old and new OLED code, but not at the same time.

**Pull request merged**